hardware_interface/src/utils/component_parser.cpp

```diff
@@ -0,0 +1,183 @@
+// Copyright 2020 ROS2-Control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <tinyxml2.h>
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+
+#include "hardware_interface/utils/component_parser.hpp"
+#include "hardware_interface/component_info.hpp"
+
+namespace
+{
+constexpr const auto kRobotTag = "robot";
```

Comment:
Is it just me, being the hungarian who doesn't like hungarian notation? We also aren't using it anywhere else on the codebase (thank god) but only for some constants. Do we even need this? I know some people don't like shouty capital-case constant but they are not that bad nor would it be horrific to simply drop the gibberish from these variable names and go with `robot_tag` in this instance...

@Karsten1987 ?

Reply:
that's the Google style guide. We've been using it in the ROS2 code base as such.

Reply:
Should I change something here?

Reply:
please just stay consistent at least. You've been using constants in three different places, all with their own style :D